include/jsn.hrl

```diff
@@ -10,36 +10,48 @@
 %% jsn types
 %%=============================================================================
 
--type json_string()      :: binary().
--type json_key()         :: json_string() | atom().
--type json_number()      :: integer() | float().
--type json_boolean()     :: true | false.
--type json_null()        :: null.
--type json_array()       :: [json_term()].
--type json_array_index() :: first | last | pos_integer().
--type json_proplist()    :: [{json_key(), json_term()}].
--type json_eep18()       :: {json_proplist()}.
--type json_struct()      :: {struct, json_proplist()}.
--type json_object()      :: json_proplist() | json_eep18() | json_struct().
-
--type json_term()        :: json_string() | json_number() | json_array() |
-                            json_object() | json_null() | json_boolean().
-
+-type json_string()        :: binary().
+-type json_key()           :: json_string() | atom().
+-type json_number()        :: integer() | float().
+-type json_boolean()       :: true | false.
+-type json_null()          :: null.
+-type json_array()         :: [json_term()].
+-type json_no_map_array() ::  [json_no_map_term()].
+-type json_array_index()   :: first | last | pos_integer().
+-type json_proplist()      :: [{json_key(), json_term()}].
+-type json_eep18()         :: {json_proplist()}.
+-type json_struct()        :: {struct, json_proplist()}.
+-type json_no_map_object() :: json_proplist() | json_eep18() | json_struct().
+-ifdef(maps_support).
+-type json_map()           :: #{json_key() => json_term()}.
+-type json_object()        :: json_no_map_object() | json_map().
+-else.
+-type json_object()        :: json_no_map_object().
+-endif.
+-type json_no_map_term()   :: json_string() | json_number() | json_no_map_array() |
+                              json_null()   | json_boolean() | json_no_map_object().
+-type json_term()          :: json_string() | json_number() | json_array() |
+                              json_null()   | json_boolean() | json_object().
 
 %% JSN OPTIONS
 %%
 %% JSN options are passed to the new/2 function when building jsn objects
 %% from scratch. Currently, the only option is format, which can be either:
 %% 
 %% * proplist (default)
```

Comment:
Yes. Changing it is a library-breaking change. it's [on the roadmap for v2.0.0](https://github.com/nalundgaard/jsn/tree/v1.0.3#200)